src/topper-web-sdk.ts

```diff
@@ -25,20 +25,33 @@ class TopperWebSdk {
   }
 
   static triggerEvent(eventName: Events, data?: any): void {
-    const mainWindow = window.self !== window.top ? window.parent : window.opener;
+    const reactNativeWebviewTarget = window.ReactNativeWebView || window.parent?.ReactNativeWebView;
 
-    if (!mainWindow) {
+    const targets = [
+      ...(reactNativeWebviewTarget ? [{ type: Targets.REACT_NATIVE_WEBVIEW, value: reactNativeWebviewTarget }] : []),
+      ...(window.opener ? [{ type: Targets.WINDOW, value: window.opener }] : []),
+      ...(window.self !== window.top ? [{ type: Targets.WINDOW, value: window.parent }] : [])
+    ];
+
+    if (targets.length === 0) {
       return;
     }
 
-    mainWindow.postMessage(
-      {
-        name: eventName,
-        payload: data,
-        source: TOPPER_WEB_SDK_EVENT_SOURCE
-      },
-      '*'
-    );
+    const message = {
+      name: eventName,
+      payload: data,
+      source: TOPPER_WEB_SDK_EVENT_SOURCE
+    };
+
+    targets.forEach(target => {
```

Comment:
np
```suggestion
    targets.forEach(({ type, value }) => {
```

or even for this for (arguably) slighter more explicit

```suggestion
    targets.forEach(({ type, value }) => {
      const payload = type === Targets.REACT_NATIVE_WEBVIEW
        ? JSON.stringify(message)
        : message;
    
      const targetOrigin = type === Targets.REACT_NATIVE_WEBVIEW
        ? undefined
        : '*';

      value.postMessage(payload, targetOrigin);
    });
```